contrib/epee/include/readline_buffer.h

```diff
@@ -19,12 +21,23 @@ namespace rdln
     }
     void get_line(std::string& line) const;
     void set_prompt(const std::string& prompt);
+    static void add_completion(const std::string& command)
+    {
+      if(std::find(completion_commands.begin(), completion_commands.end(), command) != completion_commands.end())
+	return;
+      completion_commands.push_back(command);
```

Comment:
Periodically perform a bulk erase from the front of the `std::vector` when it reaches a certain size. 
A long running daemon will continually use memory for past commands and require more CPU when scanning a command not in the history (linear search). The buffer is likely to be somewhat small so its unlikely to matter, but limiting on history is not uncommon in other applications an is a few lines of additional code.